subprojects/common/core/src/main/java/hu/bme/mit/theta/core/type/fptype/FpAddExpr.java

```diff
@@ -0,0 +1,84 @@
+package hu.bme.mit.theta.core.type.fptype;
+
+import hu.bme.mit.theta.core.model.Valuation;
+import hu.bme.mit.theta.core.type.Expr;
+import hu.bme.mit.theta.core.type.abstracttype.AddExpr;
+import hu.bme.mit.theta.core.utils.TypeUtils;
+
+import java.util.List;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static hu.bme.mit.theta.core.utils.TypeUtils.checkAllTypesEqual;
+
+public class FpAddExpr extends AddExpr<FpType> {
```

Comment:
I am not sure about AddExpr. See my comment in FpType.java

Reply:
Somewhat fixed in [2608512](https://github.com/ftsrg/theta/pull/123/commits/260851260a768790fd16a6f98dcd612f29bb0084) -- see separate discussion